packages/ds-ext/src/extension.ts

```diff
@@ -3,18 +3,39 @@ import { DiagramEditorProvider } from './DiagramEditorProvider';
 import * as fs from 'fs';
 import * as path from 'path';
 import { createDemosDirectory } from './commands/createDemosDirectory';
+import { spawn } from 'child_process';
+
+let diagramEditorProvider: DiagramEditorProvider;
 
 export function activate(context: vscode.ExtensionContext) {
   let disposable = vscode.commands.registerCommand('ds-ext.createDemos', async () => {
     createDemosDirectory();
   });
 
+  diagramEditorProvider = new DiagramEditorProvider(context);
+
   context.subscriptions.push(
     vscode.window.registerCustomEditorProvider(
       'ds-ext.diagramEditor',
-      new DiagramEditorProvider(context)
+      diagramEditorProvider
     )
   );
+
+  console.log('ds-ext is installed at ', context.extensionPath);
+  // run yarn install in the extension folder
```

Comment:
Why is this needed, please explain this a bit 🙏 Seems a bit quirky to install dependencies at activation-time?

Reply:
The reason is that the DuckDB binary is not compatible across all platforms (macOS, Windows, Linux), so users need to install DuckDB locally (this approach is based on practices from DuckDB SQL and other VSCode extensions).